craft_application/models/project.py

```diff
@@ -115,6 +137,49 @@ def effective_base(self) -> Any:  # noqa: ANN401 app specific classes can improv
             return self.base
         raise RuntimeError("Could not determine effective base")
 
+    @classmethod
+    @abc.abstractmethod
+    def _providers_base(cls, base: str) -> craft_providers.bases.BaseAlias | None:
+        """Get a BaseAlias from the application's base.
+
+        :param base: The application-specific base name.
+
+        :returns: The BaseAlias for the base or None if the base does not map to
+          a BaseAlias.
+
+        :raises CraftValidationError: If the project's base cannot be determined.
+        """
+
+    @pydantic.root_validator(  # pyright: ignore[reportUnknownMemberType,reportUntypedFunctionDecorator]
+        pre=False
+    )
+    def _validate_devel(cls, values: dict[str, Any]) -> dict[str, Any]:
+        """Validate the build-base is 'devel' for the current devel base."""
+        base = values.get("base")
+        # if there is no base, do not validate the build-base
+        if not base:
+            return values
+
+        base_alias = cls._providers_base(base)
+
+        # if the base does not map to a base alias, do not validate the build-base
+        if not base_alias:
+            return values
+
+        build_base = values.get("build_base") or base
+        build_base_alias = cls._providers_base(build_base)
+
+        for devel_base_info in DEVEL_BASE_INFOS:
```

Comment:
So I caught a bit of the conversation about supporting devel versions of centos or alma with the validator, but how is an application supposed to provide additional values in `DEVEL_BASE_INFOS`?

Reply:
After that conversation, I decided not to allow applications to provide additional values.

My rationale is that I'm not aware of a need for this today or in the near future, so I opted not to add a feature that won't be used.

If we add 'devel' bases for centos and alma in craft-providers, we can add them here.